starter-code/routes/authRoutes.js

```diff
@@ -0,0 +1,70 @@
+const express = require("express");
+const bcrypt = require("bcrypt");
+const User = require("../models/Users");
+const saltRounds = 10;
+const router = express.Router();
+
+router.get("/", (req, res, next) => {
+  res.render("index");
+});
+
+router.get("/signup", (req, res, next) => {
+  res.render("../views/signup.hbs");
+});
+
+router.get("/login", (req, res, next) => {
+  res.render("../views/login.hbs");
+});
+
+router.post("/signup", (req, res, next) => {
+  const { username, password } = req.body;
+  if (username == "" || password == "") {
+    res.render("../views/signup.hbs", {
+      error: "Username and password needed"
+    });
+  } else {
+    User.findOne({ username: username }, "username").then(data => {
+      console.log(data);
+      if (data == null) {
+        console.log("hola");
+        const salt = bcrypt.genSaltSync(saltRounds);
+        let hash = bcrypt.hashSync(username, salt);
+        userdata = { username: username, password: hash };
+        user.create(userdata).then(() => {
```

Comment:
This `user.create` was to be with CAPITAL 'U' like this `User.create(userdata).then(() => {res.redirect("/") });`
That is why the user was not creating in the DB.